lib/job-iteration/iteration.rb

```diff
@@ -150,7 +145,9 @@ def reenqueue_iteration_job
       self.times_interrupted += 1
 
       self.already_in_queue = true if respond_to?(:already_in_queue=)
-      retry_job
+
+      enqueue unless @enqueued
```

Comment:
You no-op enqueueing if `@enqueued` is set, but what about the rest of side-effects in that method? Why do they need to run if the job was already enqueued?

Reply:
Unless I'm missing something you need the `defined?(@enqueued)` otherwise you will get a Ruby warning.